sandbox/lua/modules/line_protocol.lua

```diff
@@ -0,0 +1,246 @@
+Optionally includes all standard message fields as tags or fields and
+iterates through all of the dynamic fields
+to add as points (series), skipping any fields explicitly omitted using the
+`skip_fields` config option.  It can also map any Heka message fields as tags
+in the request sent to the InfluxDB write API, using the `tag_fields` config
+option.  All dynamic fields in the Heka message are converted to separate points
+separated by newlines that are submitted to InfluxDB.
+
+API
+^^^
+
+**carbon_line_msg(config)**
+    Wrapper function that calls others within this module and the field_util
+    module to generate a table of Carbon line protocol messages that are
+    derived from the dynamic fields in a Heka message. Base fields or
+    dynamic fields can be used in the metric name portion of the message.
+    Configuration is implemented in the encoders that utilize this module.
+    This function takes no arguments.
+
+    *Arguments*
+        - config (table or nil)
+            Table of config option overrides that come from the client
+            of this API.  Defaults for this module are defined within
+            the set_config function and clients implementing this API
+            can reference it for available options.
+
+    *Return*
+        A table of Carbon line protocol messages ready to be sent to
+        a Carbon server after being looped through in an encoder implementing
+        this API.
+
+**influxdb_line_msg(config)**
+    Wrapper function that calls others within this module and the field_util
+    module to generate a table of InfluxDB line protocol messages that are
+    derived from the base or dynamic fields in a Heka message. Base fields or
+    dynamic fields can be used in the metric name portion of the message and
+    included as tags if desired.  Configuration is implemented in the encoders
+    that utilize this module.  This function takes no arguments.
+
+    *Arguments*
+        - config (table or nil)
+            Table of config option overrides that come from the client
+            of this API.  Defaults for this module are defined within
+            the set_config function and clients implementing this API
+            can reference it for available options.
+
+    *Return*
+        A table of InfluxDB line protocol messages ready to be sent to
+        an InfluxDB server after being looped through in an encoder
+        implementing this API.
+
+--]=]
+
+local interp = require "msg_interpolate"
+local field_util = require "field_util"
+local math = require "math"
+local read_config = read_config
+local read_message = read_message
+local read_next_field = read_next_field
+local pairs = pairs
+local string = require "string"
+local table = require "table"
+local tostring = tostring
+local type = type
+
+local M = {}
+setfenv(1, M) -- Remove external access to contain everything in the module.
+
+local function influxdb_kv_fmt(string)
```

Comment:
I broke this out into its own function since the substitution was made in more than two different spots in the code.